env/env_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package env
+
+import (
+	"os"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetString(t *testing.T) {
+	const expected = "foo"
+
+	assert := assert.New(t)
+
+	key := "STRING_SET_VAR"
+	os.Setenv(key, expected)
+	assert.Equal(expected, GetString(key, "~"+expected))
+
+	key = "STRING_UNSET_VAR"
+	assert.Equal(expected, GetString(key, expected))
+}
+
+func TestGetInt(t *testing.T) {
+	const expected = 1
+
+	assert := assert.New(t)
+
+	key := "INT_SET_VAR"
+	os.Setenv(key, strconv.Itoa(expected))
+	returnVal, _ := GetInt(key, 1)
```

Comment:
this doesn't prove whether it read the var or hit the default

Reply:
fixed

Reply:
Unless I misunderstand, this is still not proving.  You setenv to 1 and then read it back with a default of 1.  Which value was actually returned?  Just pass 2 here like you do in float

Reply:
sorry, updated